### PR TITLE
Force LF line endings for .ql, .qll, .qlref, and .dbscheme

### PR DESCRIPTION
**Pull request by dave-bartolomeo**
Added a .gitattributes file to specify that *.ql, *.qll, *.qlref, and *.dbscheme files should have their line endings normalized to LF in the database, and to keep the LF line endings in the working tree even on Windows. Instructions for how to renormalize files after future changes are in a comment .gitattributes.

Renormalized the existing such files in the repo. This consisted of a few .qlref files in C#, three test .ql files that @rdmarsh2 recently modified in JavaScript, and essentially every file that @dave-bartolomeo has ever edited:) I don't expect any of these files to cause merge conflicts with anything currently in flight.

I didn't know which other file extensions we want to force this for. I'm fine with forcing it for all text files, assuming that doesn't break anything.

### CODE REVIEW

**Comment by dave-bartolomeo** [comment]
Added WIP to prevent merge until all three languages sign off.

**Comment by Rae2che5** [comment]
I don't think you want to do this to the `*.dbscheme` files -- you'd trigger the need for upgrade scripts, as their content hash matters.

**Comment by pavgust** [comment]
I don't think you want to do this to the *.dbscheme files -- you'd trigger the need for upgrade scripts, as their content hash matters.

**Comment by dave-bartolomeo** [comment]
@pavgust All of the dbscheme files are already LF in the database. I've always had to be careful when I made dbscheme changes to avoid introducing rogue CRLFs. With this change, they'll remain LF in the database and in the working tree. Also, I just added a .editorconfig file to this PR, so that anyone using a supported editor will be much less likely to introduce a CRLF. Finally, we hash the dbscheme with `git hash-object`, which takes .gitattributes into account even for files not yet committed, so even if someone does add a CRLF to a dbscheme file, the computed hash will be the same as if it were normalized to LF.

**Comment by pavgust** [comment]
Ah, good -- apologies, I should have checked whether any dbscheme files actually changed. If not, then forcing their line endings to LF is indeed the right thing to do.

> Finally, we hash the dbscheme with `git hash-object`, which takes .gitattributes into account even for files not yet committed, so even if someone does add a CRLF to a dbscheme file, the computed hash will be the same as if it were normalized to LF.

There are at least some code paths that has the dbscheme in an equivalent way, but without access to the repo (and hence without reference to `.gitattributes`). Still, +1 on this PR.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[*.{ql,qll,qlref,dbscheme,}]
```
[the remark this answers is not in the document]
Comment (dave-bartolomeo) [reply .editorconfig]: Type. Fixed when I added .qhelp to the list.

**Comment by dave-bartolomeo** [comment]
I've added .qhelp files to the LF-only list. There were only two that had CRLFs.

I don't think we should do this for .expected files. We compare these using binary comparison. If someone adds CRLFs in their working tree, and the binary comparison succeeds, then the comparison would fail for everyone else. I'd rather the failure happen on the machine where the CRLF was introduced.

For other source file extensions, I'd like to hold off until we can discuss further. I'd be fine with LF for all the .js/.cs/.cpp/.html files, but I don't know if there are any of them that chose their current line endings for specific testing purposes. Also, most of these files aren't part of the distribution that end customers see, so it's less important (although still good) to have consistent line endings.

**Comment by xiemaisi** [comment]
> I'd be fine with LF for all the .js/.cs/.cpp/.html files, but I don't know if there are any of them that chose their current line endings for specific testing purposes.

Some of the alert suppression tests have CRLF line endings to test that suppression works on Windows, so indeed we'd have to be careful about enforcing LF line endings for tests.

Apparently the `plugin.xml` files also have CRLF line endings, but we can fix those up later if desired.

**Pull request merged**